.github/workflows/ci.yml

```diff
@@ -7,12 +7,64 @@ name: 💡🏠
 on: [pull_request]
 
 jobs:
-  ci:
+  # `basics` includes all non-smoke CI, except for unit and proto
+  basics:
+    runs-on: ubuntu-latest
+
+    # A few steps are duplicated across all jobs. Can be done better when this feature lands:
+    #   https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851
+    #   https://github.com/actions/runner/issues/438
+    steps:
+    - name: git clone
+      uses: actions/checkout@v2
+
+    - name: Use Node.js 10.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 10.x
 
+    - run: yarn --frozen-lockfile
+    - run: yarn build-all
+
+    # Run tests that require headfull Chrome.
+    - run: sudo apt-get install xvfb
+    - name: yarn test-clients
+      run: xvfb-run --auto-servernum yarn test-clients
+    - name: yarn test-bundle
+      run: xvfb-run --auto-servernum yarn test-bundle
+    - name: yarn test-docs
+      run: xvfb-run --auto-servernum yarn test-docs
+
+    - run: yarn diff:sample-json
+    - run: yarn type-check
+    - run: yarn lint
+    - run: yarn test-lantern
+    - run: yarn test-legacy-javascript
```

Comment:
lantern can take 1m

legacy js can take 3m

idk about lhci dogfood.

so if we want to keep it under 10 minutes always we should move these to their own job.